lib/ddtrace/contrib/httparty/request_patch.rb

```diff
@@ -0,0 +1,126 @@
+require 'ddtrace/ext/net'
+require 'ddtrace/ext/distributed'
+require 'ddtrace/propagation/http_propagator'
+require 'ddtrace/contrib/rest_client/ext'
+
+module Datadog
+  module Contrib
+    module HTTParty
+      # HTTParty RequestPatch
+      module RequestPatch
+        def self.included(base)
+          base.send(:prepend, InstanceMethods)
+        end
+
+        # InstanceMethods - implementing instrumentation
+        module InstanceMethods
+          def uri
+            # uri is a method not a property in underlying class
+            # cache it so that it is only called once per call to 'perform', and then delete it
+            @dd_uri ||= super
+          end
+
+          def perform(&block)
+            return super(&block) unless tracer.enabled
+
+            method = http_method::METHOD
+            datadog_trace_request(method, uri) do |span|
+              if distributed_tracing?
+                headers = options[:headers].respond_to?(:to_hash) ? options[:headers].to_hash : {}
+                Datadog::HTTPPropagator.inject!(span.context, headers)
+                options[:headers] = headers
+              end
+
+              super(&block)
```

Comment:
We should make sure any errors that occur in our tracing logic do not affect the application's ability to perform its normal functions.

We might want to have a `rescue` clause for all our code, and make sure that `super` is always called.

Reply:
Much of the logic within `tracer.trace` does this.